nifpga/session.py

```diff
@@ -391,7 +414,258 @@ def read(self):
+        bit. If the result after masking the the word portion of the fixed
+        point is none zero that indicates the data read has overflowed. """
+        mask = 2**(self._word_length)
+        if data & mask > 0:
+            return True
+        return False
+
+    def _remove_overflow_bit(self, data):
+        """ This helper method masks out all bits not inside the word length,
+        ultimately returning a value of data without the overflow bit. """
+        return data & (2**(self._word_length) - 1)
+
+    def _integer_twos_comp(self, data):
+        signed_bit_mask = 2**(self._word_length - 1)
+        if data & signed_bit_mask > 0:
+            data = data ^ (2 ** (self._word_length) - 1)
+            data += 1
+            data *= -1
+        return data
+
+    def write(self, user_input):
+        """ Writes the user's the users input into the register as a fixed
+        point number. Any inputs outside the bounds of this fixed point
+        register will be coerced and the user will be warned. The user input's
+        supported include any python Number.
+
+            Args:
+                Number: user numerical input to be converted to fixed point.
+                (bool, Number): Tuple with the members : Boolean for overflow,
+                                user numerical input to be converted to fixed
+                                point.
+        """
+
+        fxp_representation = self._convert_user_input_to_fxp_representation(user_input)
+        arrayData = self._extract_array_of_u32_from_fxp_value(fxp_representation)
+        buf = self._ctype_type(*arrayData)
+        self._write_func(self._session, self._resource, buf, self._transfer_len)
+
+    def _convert_user_input_to_fxp_representation(self, user_input):
+        (overflow, data) = self._validate_and_parse_user_input(user_input)
+
+        fxp_representation = 0
+        if data < self._minimum:
+            fxp_representation = self._convert_value_to_fxp(self._minimum)
+            self.warn_coerced_data()
+        elif data > self._maximum:
+            fxp_representation = self._convert_value_to_fxp(self._maximum)
+            self.warn_coerced_data()
+        else:
+            fxp_representation = self._convert_value_to_fxp(data)
+
+        if self._signed and data < 0:
+            fxp_representation = self._integer_twos_comp(fxp_representation)
+
+        if self._overflow_enabled:
+            if overflow:
+                fxp_representation += 2**(self._word_length)
+
+        return fxp_representation
+
+    def _validate_and_parse_user_input(self, user_input):
+        overflow = None
+        data = None
+        if self._overflow_enabled:
+            try:
+                (overflow, data) = user_input
+            except TypeError:
+                overflow = False
+                data = user_input
+                warn("Using default value 'False' for overflow status.")
```

Comment:
Sorry I wasn't clear in my original comment.  I would say don't warn here.  

I think its a completely fine use case if the customer doesn't think about the overflow bit in the write case.  The warning might hint to them they are they sending unnecessary data, but what they're doing is still completely fine and would take an FPGA recompile to remove.

Reply:
Fixed.